analytic_engine/src/storage_options.rs

```diff
@@ -119,3 +120,38 @@ impl ObkvOptions {
         8
     }
 }
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct S3Options {
+    pub region: String,
+    pub key_id: String,
+    pub key_secret: String,
+    pub endpoint: String,
+    pub bucket: String,
+    pub prefix: String,
+    #[serde(default = "S3Options::default_pool_max_idle_per_host")]
```

Comment:
Extract fields in  `L131-L138` into a HttpOptions, and S3/Aliyun use it to share code.

Reply:
Refer to https://serde.rs/attr-flatten.html#factor-out-frequently-grouped-keys to ensure all the keys flattened.